### DISCUSSION TITLE
Support for HIP architecture

### DISCUSSION THREAD

**Discussion by b-rousse** [ACEsuit/mace#546] [Ideas]
Hi all, 

I was wondering if there is support for HIP architecture in existence or on the horizon? 

If not, would the conversion be relatively straightforward, or does the code use CUDA primitives?

Thanks,
Ben

**Comment by ilyes319** [comment]
Not sure, what kind of hardware is that? If Pytorch supports it, or you have an extension to make a pytorch device of it, then it should be fine. You can also try to compile a model to a low level language that is compatible with it using torch.compile.

**Reply by gabor1** [reply]
I think it's AMD stuff

**Reply by b-rousse** [reply]
Yes it's AMD's analog for CUDA, and was designed with portability from NVIDIA GPUs in mind. The only thing that would complicate porting MACE to HIP support would be if CUDA primitives were used in the development of MACE.

**Reply by ilyes319** [reply]
Only pytorch is used in this repo, so it should work hopefully.

**Reply by vishank-u** [reply]
Can someone confirm if this worked with AMD's HIP architecture? If yes, with which version of pytorch ROCm?

**Reply by b-rousse** [reply]
@vishank-u I got it to work on a single GPU setup using ROCm 5.7.1 and libtorch 2.2.2. 

The issue I'm facing now is getting the MACE pair style to perform correctly with MPI-aware GPU.
